osf/models/provider.py

```diff
@@ -219,6 +220,14 @@ class RegistrationProvider(AbstractProvider):
 
     DEFAULT_SUBSCRIPTIONS = ['new_pending_submissions', 'new_pending_withdraw_requests']
 
+    # A list of dictionaries describing new fields that providers want to surface on their registrations
+    # Each entry must provide a 'field_name' key. In the future, other keys may be supported to enable
+    # better i18n, validation, or other helpful things.
+    #
+    # Ex:
+    # [{'field_name': 'foo'}, {'field_name': 'bar'}]
+    additional_metadata_fields = DateTimeAwareJSONField(blank=True)
```

Comment:
You should make this default to list, or you can actually define this field as an ArrayList of JSON (I believe) that should validate better, less JSON-blob-y.

Reply:
Removed the defaults on the keys to speed up the migration, though this one is almost certainly less costly than the one on Registration. Will look into the ArrayField option, though StackOverflow doesn't appear to think too kindly about mixing ArrayField with complex base field types on first glance.